toolchains/lowrisc_toolchain_rv32imc/clang_wrappers/nix/cpp

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# MIT License
+#
+# Copyright (c) 2019 Nathaniel Brough
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+set -euo pipefail
+# Workaround to replace all system includes with user includes
+external/com_lowrisc_toolchain_rv32imc_compiler/bin/riscv32-unknown-elf-clang++ "${@}" -fdiagnostics-color=always
+#external/com_lowrisc_toolchain_rv32imc_compiler/bin/riscv32-unknown-elf-cpp "$@"
```

Comment:
It appears in the `bzl` files that the word `cpp` may be used inconsistently.  In this context, invoking `cpp` means the C preprocessor.

If you were to write an exemplary hello world program and then ran `cpp` on it, it would emit your source code with `stdio.h` included and line number/filename directives for the next stage of compilation.

In the bazel-embedded project, the x64 clang wrappers simply invoke `clang++` withou any options, so its not clear whether or not bazel ever really invoked just the preprocessor, but it seems to at least want the preprocessor listed in the set of tools.